compiler/rustc_mir_transform/src/const_prop_lint.rs

```diff
@@ -515,16 +541,212 @@ impl<'mir, 'tcx> ConstPropagator<'mir, 'tcx> {
 
     fn ensure_not_propagated(&self, local: Local) {
         if cfg!(debug_assertions) {
+            let val = self.get_const(local.into());
             assert!(
-                self.get_const(local.into()).is_none()
+                matches!(val, Some(Value::Uninit))
                     || self
                         .layout_of(self.local_decls()[local].ty)
                         .map_or(true, |layout| layout.is_zst()),
-                "failed to remove values for `{local:?}`, value={:?}",
-                self.get_const(local.into()),
+                "failed to remove values for `{local:?}`, value={val:?}: {:#?}",
+                self.body,
```

Comment:
Please don't print the body here. For debugging, we should use `-Zdump-mir`.

Reply:
oh whoops, that was just a temporary helper for myself